GDAXSharp/WebSocket/WebSocket.cs

```diff
@@ -0,0 +1,139 @@
+﻿using GDAXSharp.Network.Authentication;
+using GDAXSharp.Shared.Types;
+using GDAXSharp.Shared.Utilities.Clock;
+using GDAXSharp.Shared.Utilities.Extensions;
+using GDAXSharp.WebSocket.Models.Request;
+using GDAXSharp.WebSocket.Models.Response;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using WebSocket4Net;
+
+namespace GDAXSharp.WebSocket
+{
+    public class WebSocket : IWebSocket
+    {
+        private const string ApiUri = "wss://ws-feed.gdax.com";
+
+        private const string SandBoxApiUri = "wss://ws-feed-public.sandbox.gdax.com";
+
+        private readonly IAuthenticator _authenticator;
+
+        private readonly IClock _clock;
+
+        private readonly bool _sandBox;
+
+        private CancellationToken CancellationToken { get; set; }
+        private ProductType[] ProductTypes { get; set; }
+        private WebSocket4Net.WebSocket WebSocketFeed { get; set; }
+
+        public WebSocket(IAuthenticator authenticator, IClock clock, CancellationToken cancellationToken, bool sandBox = false)
```

Comment:
i don't think the user should have to pass in an `IClock`. they shouldn't care about it imo